pkg/volume/flexvolume/attacher.go

```diff
@@ -63,9 +63,13 @@ func (a *flexVolumeAttacher) WaitForAttach(spec *volume.Spec, devicePath string,
 
 // GetDeviceMountPath is part of the volume.Attacher interface
 func (a *flexVolumeAttacher) GetDeviceMountPath(spec *volume.Spec) (string, error) {
-	mountsDir := path.Join(a.plugin.host.GetPluginDir(flexVolumePluginName), a.plugin.driverName, "mounts")
 
-	return (*attacherDefaults)(a).GetDeviceMountPath(spec, mountsDir)
+	volumeName, err := a.plugin.GetVolumeName(spec)
```

Comment:
GetVolumeName() is called before every makeGlobalMountPath(). Maybe move the call inside makeGlobalMountPath().